controllers/cluster_controller.go

```diff
@@ -241,3 +233,20 @@ func (r *ClusterReconciler) listChildren(ctx context.Context, cluster *clusterv1
 
 	return children, nil
 }
+
+func (r *ClusterReconciler) patchCluster(ctx context.Context, cluster *clusterv1.Cluster, patch client.Patch) error {
+	log := r.Log.WithValues("cluster-namespace", cluster.Namespace, "cluster-name", cluster.Name)
+	// TODO(chuckha): .Patch will use the typed client and in the case of a noop on the Spec will return an empty status which is automatically assigned to cluster.
+	// This means that Status.Patch will not update anything since all new Status data was eliminated.
+	statusCopy := cluster.Status.DeepCopy()
+	if err := r.Client.Patch(ctx, cluster, patch); err != nil {
+		log.Error(err, "Error patching cluster")
+		return err
+	}
+	statusCopy.DeepCopyInto(&cluster.Status)
+	if err := r.Client.Status().Patch(ctx, cluster, patch); err != nil {
```

Comment:
I wonder what happens if we patch Status first? Would the same thing happen?

Reply:
In that case it seems the cluster metadata/resource version is not updated and we'd have to call Get on the cluster object again before saving the spec

Reply:
I suspect it is likely if there are only non-Status changes